tests/ui/missing_const_for_fn/cant_be_const.rs

```diff
@@ -32,11 +32,8 @@ fn random_caller() -> u32 {
 
 static Y: u32 = 0;
 
-// We should not suggest to make this function `const` because const functions are not allowed to
-// refer to a static variable
 fn get_y() -> u32 {
```

Comment:
This shouldn't be linted still. Adding `const` to this function gives the error described in the comment. But for some reason `rustc_mir::transform::qualify_min_const_fn::is_min_const_fn` returns `Ok(())` for this function:

https://github.com/rust-lang/rust-clippy/blob/b4f1769734b6204fc6bece8556b7b80a7683271e/clippy_lints/src/missing_const_for_fn.rs#L108-L110

cc @oli-obk Does `optimized_mir` replaces the static with a const value and `is_min_const_fn` then thinks this could be a const fn?

Reply:
Did something change in the handling of `Statement`s in MIR recently? The error in `redundant_clone` is similar. This code
https://github.com/rust-lang/rust-clippy/blob/b4f1769734b6204fc6bece8556b7b80a7683271e/clippy_lints/src/redundant_clone.rs#L374-L383

stopped detecting, that in `(a.clone(), a)` both `a`s are the same.

Reply:
What I mean with

> The error [...] is similar

is, that in both cases the code that should check for them is never actually reached.

Reply:
oh... we run is_min_const_fn on the optimized mir while rustc runs it on `mir_const`. This will keep causing problems. One thing clippy could do is to make the `optimized_mir` query not do any optimizations. This is possible since https://github.com/rust-lang/rust/pull/66297

Reply:
Naive question: Could we also just run `is_min_const_fn` on `mir_const` instead of `optimized_mir`?

Reply:
So how do I override the `optimized_mir` query? When I just copy and paste its source code [1] from rustc and remove the `run_optimization_passes` call¹, a lot of tests began to ICE. I added this:
```rust
        config.override_queries = Some(|_, providers, _| {
            providers.optimized_mir = |tcx, def_id| {
                // (Mir-)Borrowck uses `mir_validated`, so we have to force it to
                // execute before we can steal.
                tcx.ensure().mir_borrowck(def_id);

                let (body, _) = tcx.mir_validated(def_id);
                let body = body.steal();
                tcx.arena.alloc(body)
            }
        });
```

to 
https://github.com/rust-lang/rust-clippy/blob/b4f1769734b6204fc6bece8556b7b80a7683271e/src/driver.rs#L63-L64

---
¹ I also removed this:
```rust
    if tcx.is_constructor(def_id) {
        // There's no reason to run all of the MIR passes on constructors when
        // we can just output the MIR we want directly. This also saves const
        // qualification and borrow checking the trouble of special casing
        // constructors.
        return shim::build_adt_ctor(tcx, def_id);
    }
```

because the `shim` module is private.

---
[1] https://github.com/rust-lang/rust/blob/9d6ff1553b7debbe5c99c102ce0978b6130592f8/src/librustc_mir/transform/mod.rs#L316-L333

Reply:
Hmm. So we'd need to make more things public. But that's probably not the failure reason. One thing you can try is set the mir opt level to 0 instead of trying to replace the query. If that doesn't work, reintroduce the mir optimizations but remove all the ones that break clippy (e.g. const_prop)

Reply:
> One thing you can try is set the mir opt level to 0

How do I set the opt level? And how can I enable/disable mir passes?